### ISSUE TITLE
[BUG] Native pinch and zoom becomes unreliable on Android Chrome

### ISSUE THREAD

**Issue by radekropka** [framer/motion#509]
**1. Describe the bug**
Using native browser's pinch and zoom becomes unreliable when framer-motion is added to the page. Sometimes pinch and zoom works, sometimes not.

This is caused by global touch event listener added by framer-motion to the document:
```
document.addEventListener("touchmove", function (event) {
    if (isViewportScrollBlocked) {
        ...
```

**2. CodeSandbox**
[https://codesandbox.io/s/new-meadow-br2o0](https://codesandbox.io/s/new-meadow-br2o0)

**3. Steps to reproduce**
1. Use Chrome on Android browser.
2. Go to [https://br2o0.csb.app/](https://br2o0.csb.app/).
3. Pinch and zoom.
4. Notice that pinch and zoom is not always working.

**4. Expected behavior**
Pinch and zoom is always working.

**5. Video of unreliable pinch and zoom**
![withListener](https://user-images.githubusercontent.com/38790394/79476661-2a3a2e00-8001-11ea-8d7b-d33ac080f1b7.gif)

**6. The same page with event listener removed using dev tools**
![noListener](https://user-images.githubusercontent.com/38790394/79477104-afbdde00-8001-11ea-97b4-559890a1b9d5.gif)

**7. Chrome browser highlights scrolling performance issues**
![scrolling](https://user-images.githubusercontent.com/38790394/79477649-5b672e00-8002-11ea-97a6-f62e355d6107.png)

**8. Environment details**
Chrome 81
Android 9
Sony Xperia XA2 (H3113)

**Comment by lemon-juice** [comment]
I don't use this library but I'll add my 2 cents because I've stumbled across the same problem when developing a custom photo swipe js code for a web site and after long searching you seem to be the only person that has discovered and described this problem so far. After some tests I've come to the conclusion that registering touchmove as not passive is the problem:

`element.addEventListener('touchmove', function, {passive: false});`

It is definitely a Chrome (Chromium) bug since the same code in Firefox doesn't cause pinch zoom problems. I've seen it on Nokia 5.1 Plus and on an older LG phone. Opera is also affected.

This can be pretty annoying but I don't know if anything can be done. A workaround is to remove the 'passive: false' option but then preventDefault() will not work, which is not always an option.

Even if the browser default pinch zoom gesture happens not to work the touchstart and touchmove events are correctly triggered for both fingers so another workaround might be to cancel the browser pinch completely with preventDefault() and implement your own pinch zoom gesture in js that will work on the image or another element - but it may not always be what you are after and it's unnecessary work of reinventing the wheel.

I think this bug should be reported to the Chrome devs .

**Comment by lemon-juice** [comment]
I have some further observations - Chrome exhibits the pinch zoom problem if any kind of touchmove event is registered on an element regardless of the passive option. So even this:

`element.addEventListener('touchmove', function);`

will cause Chrome to sometimes get stuck when pinch zooming. However, this:

`document.addEventListener('touchmove', function);`

will not be problematic. But this will:

`document.addEventListener('touchmove', function, {passive: false});`

In other words, if the event is on the whole document then it is possible to prevent the problem by not using `{passive: false}`. However, on other elements this is not the case.

In one implementation where I wanted the zoom to work reliably I had to register the event on the whole document instead of just one element. This doesn't make sense, but that's the nature of the bug, AFAIK.

**Comment by o-alexandrov** [comment]
If not setting `passive: true` was intentional, it'd be great to know the reasons.
Otherwise, enabling `passive` would be greatly beneficial for examples like:
- Slideshow (with route changes)

**Comment by mattgperry** [comment]
The global event listener has since been removed, if there's remaining related bugs please reopen with a sandbox on the latest version.
